crates/extrinsics/src/instantiate.rs

```diff
@@ -377,7 +455,10 @@ impl Exec {
     }
 
     /// Dry run the instantiation before tx submission. Returns the gas required estimate.
-    async fn pre_submit_dry_run_gas_estimate(&self) -> Result<Weight> {
+    pub async fn pre_submit_dry_run_gas_estimate(
```

Comment:
Is this method necessary to have in call and instantiate?

I'm wondering, it should be possible to just submit the dry run to get the estimated gas regardless of it's an instantiation or call right.